src/controller/ReadInteraction.h

```diff
@@ -45,11 +45,11 @@ template <typename DecodableAttributeType>
 CHIP_ERROR ReportAttribute(Messaging::ExchangeManager * exchangeMgr, EndpointId endpointId, ClusterId clusterId,
                            AttributeId attributeId, ReportAttributeParams<DecodableAttributeType> && readParams)
 {
-    app::AttributePathParams attributePath(endpointId, clusterId, attributeId);
     app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
     CHIP_ERROR err                       = CHIP_NO_ERROR;
 
-    readParams.mpAttributePathParamsList    = &attributePath;
+    readParams.mpAttributePathParamsList    = new app::AttributePathParams[1];
```

Comment:
Platform::New please.

Reply:
And wrap it in a unique_ptr to prevent leaks.

Reply:
Done, thanks